feat(dlp): Add code samples to de-identify files in Cloud Storage

Avani-Thakker-Crest opened the pull request:
## Description
Implemented dlp_deidentify_cloud_storage code sample and test is implemented using mocking mechanism. Json equivalent: https://cloud.google.com/dlp/docs/deidentify-storage#code-example

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

m-strzelczyk commented:
Hey @Avani-Thakker-Crest this file is growing to quite a big number of lines. Would it be possible to slip it into multiple files? So that for example each region tag gets its own file?

It would also allow you to remove some `# noqa:` comments, as you wouldn't repeat imports etc.

Avani-Thakker-Crest commented:
> > Hey @Avani-Thakker-Crest this file is growing to quite a big number of lines. Would it be possible to slip it into multiple files? So that for example each region tag gets its own file?
> > It would also allow you to remove some `# noqa:` comments, as you wouldn't repeat imports etc.
> 
> Thank you @m-strzelczyk for your comment. Definitely, we can split into multiple files. Please let me know how should I split:
> 
> 1. Split deid related samples into 3-4 logical files. I had already split deid samples into two files in the past a) deid_table.py containing all samples deidentifying table data and b) deid.py containing the remaining samples. We can further go ahead and split into 2 more files.
> 2. Split samples such a way that one file corresponds to one sample. This would result in entire restructuring of all the dlp samples, be it inspect.py, risk.py and so on to maintain uniformity.
> 3. For now, we can create a separate file for deidentify_cloud_storage sample and a corresponding test file.
> 
> Please let me know which option would be appropriate to perform. Thank you!

Hi @m-strzelczyk , for now I have created new files for this particular sample. And for the rest samples, I have created a backlog issue https://github.com/GoogleCloudPlatform/python-docs-samples/issues/10601 .

m-strzelczyk commented on dlp/snippets/deid_cloud_storage.py:
```diff
@@ -0,0 +1,234 @@
+
+        # Sleep for a short duration before checking the job status again.
+        time.sleep(30)
+        no_of_attempts -= 1
```
This can go into negatives, why do we have this variable?

Avani-Thakker-Crest replied:
Thank you for your comment, this is my miss, I have updated things now.

m-strzelczyk commented on dlp/snippets/deid_cloud_storage.py:
```diff
@@ -0,0 +1,234 @@
+        job = dlp.get_dlp_job(request={"name": job_name})
+
+    if job.state != google.cloud.dlp_v2.DlpJob.JobState.DONE:
+        print(f"Job did not complete within {timeout} minutes.")
```
I think you need to fix the loop above a bit. Currently, there are only 2 stop conditions: the job is DONE or it FAILED. If it failed, we return, so we won't land here.

Avani-Thakker-Crest replied:
Thank you for your comment, this is my miss, I have updated things now.

m-strzelczyk commented on dlp/snippets/deid_cloud_storage_test.py:
```diff
@@ -0,0 +1,78 @@
+    assert "Processed Bytes" in out
+    assert "Info type: EMAIL_ADDRESS" in out
+
+    mock_dlp_instance.create_dlp_job.assert_called_once()
+    mock_dlp_instance.get_dlp_job.assert_called_once()
```
Could you perhaps add some code checking the arguments the methods were called with?

Avani-Thakker-Crest replied:
Updated.

m-strzelczyk reviewed (changes requested):
We're getting close :)

One issue with the logic inside the sample, probably easy to fix.

And one issue with increasing the depth of testing.
